packages/wallet/src/redux/protocols/concluding/instigator/container.tsx

```diff
@@ -44,25 +42,38 @@ class ConcludingContainer extends PureComponent<Props> {
           />
         );
       case 'ConcludingInstigator.WaitForOpponentConclude':
-        return <WaitForOpponentConclude />;
+        return <WaitForOtherPlayer name={'conclude'} channelId={state.channelId} />;
       case 'ConcludingInstigator.AcknowledgeConcludeReceived':
         return (
-          <ApproveDefunding
-            approve={() => defund(processId, state.channelId)}
-            keepOpen={() => keepOpen({ processId })}
+          <ApproveX
+            title={'Channel concluded'}
+            description={'Do you want to keep your ledger channel open, or defund it?'}
+            yesMessage={'Keep open'}
+            noMessage={'Defund'}
+            rejectionAction={() => defund(processId, state.channelId)}
+            approvalAction={() => keepOpen({ processId })}
           />
         );
       case 'ConcludingInstigator.ApproveConcluding':
         return (
-          <ApproveConcluding
-            deny={() => deny({ processId })}
-            approve={() => approve({ processId })}
+          <ApproveX
+            title={'Approve concluding this channel'}
+            description={'Do you want to conclude this channel? This action is not reversible'}
+            yesMessage={'Conclude'}
+            noMessage={'Cancel'}
+            rejectionAction={() => deny({ processId })}
+            approvalAction={() => approve({ processId })}
           />
         );
       case 'ConcludingInstigator.WaitForLedgerUpdate':
         return <ConsensusUpdate state={state.consensusUpdateState} />;
       case 'ConcludingInstigator.WaitForOpponentSelection':
-        return <WaitForOpponentDecision />;
+        return (
+          <WaitForOtherPlayer
+            name={'decision about defunding the ledger channel'}
```

Comment:
Even though `WaitForTheOtherPlayer` was not created as part of this PR, I am leaving this comment. Is there a better property description than `name`? `action-descriptor`?

Reply:
:+1: